src/main/java/teammates/ui/controller/InstructorFeedbackAbstractAction.java

```diff
@@ -0,0 +1,168 @@
+package teammates.ui.controller;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+
+import teammates.common.datatransfer.CourseAttributes;
+import teammates.common.datatransfer.FeedbackSessionAttributes;
+import teammates.common.datatransfer.FeedbackSessionType;
+import teammates.common.datatransfer.InstructorAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.EmailType;
+import teammates.common.util.Sanitizer;
+import teammates.common.util.TimeHelper;
+
+public abstract class InstructorFeedbackAbstractAction extends Action {
+
+    /**
+     * Common method to Get the feedback data
+     */
+    protected FeedbackSessionAttributes extractFeedbackSessionDataHelper(FeedbackSessionAttributes newSession) {
+
+        return this.extractFeedbackSessionDataHelper(newSession);
+    }
+
+    protected FeedbackSessionAttributes extractFeedbackSessionDataHelper(
+            FeedbackSessionAttributes newSession, List<String> sendRemainderEmailsList) {
+        return this.extractFeedbackSessionDataHelper(newSession, sendRemainderEmailsList);
```

Comment:
> It affects the code readability and changes will not be minimum.

In what way does having just one method affect code readability?
Having two functions with vague names (`*Helper`), overloaded but doing entirely different things (overloaded functions should do similar things), now that affects readability very negatively.

> i wasn't sure whether the order of attributes creation matters but now due to the failing tests and the build log(commit 581f184) it matters.

Quoting what I said before: failing tests is not a good reason. "It matters": in what ways do it matter?

> In the method extractFeedbackSessionData why do we have to send the reminder email about the feedback to all ?, we need to send it to only those who haven't submitted the feedback. If we create the List<String> and String[] at starting this happens.

I don't get what this means.

> there are some attributes like setCreatedTime which has to be created in the start. Remaining attributes which uses time depends on this.

Which lines give you this impression? I am pretty sure this is not the case, although I might be wrong.

Also, do address the first two items of my very first comment.